app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/CreateGroup.java

```diff
@@ -133,19 +145,48 @@ else if(groupPreference.equals("Males Only Group"))
         Group groupTemplate = new Group(groupName, courseCode, groupColor, preferenceSelected,
                 getUserName(), groupSizeMax);
 
-        groupRef.add(groupTemplate).addOnCompleteListener(new OnCompleteListener<DocumentReference>() {
+
+
+//        groupRef.add(groupTemplate).addOnCompleteListener(new OnCompleteListener<DocumentReference>() {
+//            @Override
+//            public void onComplete(@NonNull Task<DocumentReference> task) {
+//                if(task.isSuccessful()){
+//                    groupId = task.getResult().getId();
+//                    Toast.makeText(CreateGroup.this, "Group ID: " + groupId, Toast.LENGTH_SHORT).show();
+//                }
+//                else {
+//                    Toast.makeText(CreateGroup.this, task.getException().toString(), Toast.LENGTH_SHORT).show();
+//                }
+//            }
+//        });
+//        groupRef.add(groupTemplate);
+        groupId = FirebaseFirestore.getInstance().collection("groups").document().getId();
```

Comment:
I usually have a global variable called db that stores the instance of firestore. that way  i instantiate it at the beginning and call it wherever i need it in the file instead of doing FirebaseFirestore.getInstance() every time.

Reply:
Adjusted.